From 6a698feedd45fcfd0e40950cff3ff020343bb2ed Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Mon, 2 Nov 2015 11:41:02 +0800
Subject: [PATCH 0671/5242] MLK-11803 pci: imx: imx6qp pcie ep self-test
 failed

commit  f8f3e0940773917003d33970d9f48782de9302c8 from
https://source.codeaurora.org/external/imx/linux-imx.git

imx pcie used the wrab mode to do the cached access
methods on axi bus. There is 64bytes address mis-aligned
problem.
Disable the cached operations.

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c |    6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index 2cbc470..653a715 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -1360,11 +1360,7 @@ static int __init imx6_pcie_probe(struct platform_device *pdev)
 		 * reserved ddr memory after write the ddr_test_region
 		 * content to rc.
 		 */
-		if (imx6_pcie->variant == IMX7D)
-			pcie_arb_base_addr = ioremap_nocache(pp->mem_base,
-					test_region_size);
-		else
-			pcie_arb_base_addr = ioremap_cache(pp->mem_base,
+		pcie_arb_base_addr = ioremap_nocache(pp->mem_base,
 					test_region_size);
 
 		if (!pcie_arb_base_addr) {
-- 
1.7.9.5

